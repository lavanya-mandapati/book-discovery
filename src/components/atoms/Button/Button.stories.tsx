import React from 'react';
import CustomButton from './Button';

export default {
    title: 'atoms/Button',
    component: CustomButton
};

const TemplateA = (args) => <CustomButton name="Start Your Journey" />;

const TemplateB = (args) => <CustomButton name="Reply" />;

export const Primary = TemplateA.bind({});

export const Basic = TemplateB.bind({});

Basic.args = {
    variant: "outlined",
};

Primary.args = {
    color: "primary",
    variant: "contained",
};
