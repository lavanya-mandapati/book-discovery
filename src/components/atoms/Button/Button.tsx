import React from "react";
import {Button as MuiButton, Typography} from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import {COLORS} from '../../../theme/constants';


interface ButtonProps {
  name: string;
  onClick?: () => void;
  variant?: "contained"|"outlined";
  size?: string;
}
const useStyles = makeStyles({
  root: {
  width: 'auto',
  height: '38px',
  padding: '8px 21px 8px 23px',
  borderRadius: '4px',
  border: 'solid 1px COLORS.TRANSPARENT_0',
  textTransform: "none",
  color: COLORS.WHITE_50,
  backgroundColor: COLORS.PRIMARY_COLOR_500
}
});

const Button: React.FC<ButtonProps> = ({ 
    name,
    children,
    onClick, 
    variant="outlined"
  }) => { 
  const classes = useStyles();
  return (
    <MuiButton className={classes.root} variant={variant} onClick={()=> onClick}  >
     <Typography variant="body1" >{name}</Typography>
    </MuiButton>
  );
}

export default Button;