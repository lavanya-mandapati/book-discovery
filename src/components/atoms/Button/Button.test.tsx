import {render} from "@testing-library/react";
import Button from './Button';

describe("button renders correctly", () => {
    test("should match snapshot", () => {
        const { asFragment } = render(<Button name="Discover" />);
        expect(asFragment()).toMatchSnapshot()
    });
  
  });

