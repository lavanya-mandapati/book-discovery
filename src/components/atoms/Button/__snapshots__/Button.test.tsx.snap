// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button renders correctly check the button name 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-1 MuiButton-containedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Discover
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`button renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-1"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Discover
      </p>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
